src/HigherOrderExpectation.php

```diff
@@ -63,6 +63,16 @@ public function not(): HigherOrderExpectation
         return $this;
     }
 
+    /**
+     * Creates a new expectation.
+     *
+     * @param mixed $value
+     */
+    public function and($value): HigherOrderExpectation
+    {
+        return new self($this->expect($value), $value);
```

Comment:
I would expect this to be just `return new Expectation($value);`. Can you explain me your reasoning?

Reply:
I just looooove overcomplicating things. Good catch. Updated. Also allows us to be more specific with Generics.